aten/src/ATen/functorch/TensorWrapper.cpp

```diff
@@ -38,19 +38,27 @@ void dumpTensor(std::ostream& ss, const Tensor& tensor) {
 }
 
 void TensorWrapper::refreshMetadata() {
-  auto dim = value_.dim();
-  auto sizes = value_.sizes();
-  auto strides = value_.strides();
-  storage_offset_ = value_.storage_offset();
-  sizes_and_strides_.resize(value_.dim());
-  for (int64_t i = 0; i < dim; i++) {
-    sizes_and_strides_.size_at_unchecked(i) = sizes[i];
-    sizes_and_strides_.stride_at_unchecked(i) = strides[i];
+  // update size, strides and storage_offset
+  if (value_.unsafeGetTensorImpl()->has_symbolic_sizes_strides()) {
+    // for tensor with symbolic size and strides
+    set_sizes_and_strides(
+        value_.sym_sizes(), value_.sym_strides(), value_.sym_storage_offset());
+  } else {
+    // for tensor with regular size and strides
```

Comment:
Do we actually need this branch? IIRC `value_.sym_sizes()` is a generalization of `sizes()` that works even if the Tensor isn't symbolic

Reply:
I tried without this branch and all the tests trip at these INTERNAL_ASSERT

https://github.com/pytorch/pytorch/blob/6c9b94dcda696984552769da6cf578ccca0cfbdd/aten/src/ATen/functorch/TensorWrapper.cpp#L162-L172